napari_animation/_qt/animationslider_widget.py

```diff
@@ -14,19 +15,34 @@ class AnimationSliderWidget(QSlider):
     ----------
     animation : napari_animation.Animation
         napari-animation animation in sync with the GUI.
+    interpol_states : dict
+        Dictionary of interpolated states
+    keyframe_to_frame : array
+        Frame numbers corresponding to each key frames.
+    requires_update : bool
+        Define if interpolated states should be updated.
     """
 
     def __init__(self, animation, orientation=Qt.Horizontal, parent=None):
         super().__init__(orientation=orientation, parent=parent)
 
         self.animation = animation
+        self.interpol_states = []
+        self.keyframe_to_frame = np.array([])
         self.requires_update = True
+
         self.setToolTip("Scroll through animation")
 
     def _compute_states(self):
         """Computer interpolation states"""
         self.interpol_states = []
         for state in self.animation._state_generator():
             self.interpol_states.append(state)
+
+        self.keyframe_to_frame = [0]
+        steps = [keyframe['steps'] for keyframe in self.animation.key_frames]
+        self.keyframe_to_frame += np.cumsum(steps)
+
         self.setMaximum(len(self.interpol_states)-1)
+
         self.requires_update = False
```

Comment:
Thanks ! :) I sometimes leave these for myself, as I find it more readable, but it wasn't meant to stay indeed ! By the way, is there a plan to integrate a `pre-commit` in the project ?

Reply:
We haven't discussed it but we definitely should! I'll open up an issue, if you'd like to tackle this that would be really useful 🙂

Reply:
Yes, I think we should. Contribution would be very welcome

Reply:
Cool ! I added a setup.cfg file that I've taken from `napari`, but commented metadata and setup `isort` to work with `black`. There might some things to add/modify of course, but the commits pass these tests !

Reply:
@Fifourche do you mind if we add the pre-commit in it's own PR, and you remove the copied `setup.cfg`. It will make merging this PR easier and keep our git history a little cleaner. I'd like us to find a way to add pre-commit without having to add the whole napari setup.cfg. Thanks!!

Reply:
I get it, will do ! :)